File: vciplreportsv1001/vcipl1001/report/monthwises_purchase/monthwises_purchase.js
frappe.query_reports["Monthwises Purchase"] = {

    filters: [
        {
            fieldname: "supplier_group",
            label: "Supplier Group",
            fieldtype: "Link",
            options: "Supplier Group"
        },
        {
            fieldname: "month",
            label: "Month",
            fieldtype: "Select",
            options: [
                "",
                "January", "February", "March", "April", "May", "June",
                "July", "August", "September", "October", "November", "December"
            ]
        }
    ],

    after_datatable_render: function (report) {

        $(".supplier-link").off("click").on("click", function (e) {
            e.preventDefault();

            const supplier = $(this).data("supplier");
            const filters = report.get_values();

            frappe.call({
                method: "vciplreportsv1001.vciplreportsv1001.vcipl1001.report.monthwises_purchase.monthwises_purchase.get_month_breakup",
                args: {
                    supplier: supplier,
                    supplier_group: filters.supplier_group,
                    month: filters.month
                },
                callback: function (r) {

                    const content = r.message.html + `
                        <div id="supplier_purchase_chart" style="margin-top:20px;"></div>
                    `;

                    frappe.msgprint({
                        title: "Month-wise Purchase - " + supplier,
                        message: content,
                        wide: true
                    });

                    setTimeout(() => {
                        new frappe.Chart("#supplier_purchase_chart", {
                            title: "Monthly Purchase Trend",
                            data: {
                                labels: r.message.labels,
                                datasets: [{
                                    name: "Purchase Amount",
                                    type: "bar",
                                    values: r.message.values
                                }]
                            },
                            type: "bar",
                            height: 300
                        });
                    }, 300);
                }
            });
        });
    }
};

File: vciplreportsv1001/vcipl1001/report/monthwises_purchase/monthwises_purchase.py
import frappe


@frappe.whitelist()
def get_month_breakup(supplier=None, supplier_group=None, month=None):
    conditions = []
    values = {}

    if supplier:
        conditions.append("pi.supplier = %(supplier)s")
        values["supplier"] = supplier

    if supplier_group:
        conditions.append("s.supplier_group = %(supplier_group)s")
        values["supplier_group"] = supplier_group

    if month:
        conditions.append("MONTHNAME(pi.posting_date) = %(month)s")
        values["month"] = month

    condition_sql = " AND ".join(conditions)
    if condition_sql:
        condition_sql = " AND " + condition_sql

    data = frappe.db.sql(f"""
        SELECT
            MONTHNAME(pi.posting_date) AS month,
            SUM(pii.amount) AS total
        FROM `tabPurchase Invoice` pi
        INNER JOIN `tabPurchase Invoice Item` pii ON pii.parent = pi.name
        INNER JOIN `tabSupplier` s ON s.name = pi.supplier
        WHERE pi.docstatus = 1
        {condition_sql}
        GROUP BY MONTH(pi.posting_date)
        ORDER BY MONTH(pi.posting_date)
    """, values, as_dict=True)

    labels = [d.month for d in data]
    values = [d.total for d in data]

    html = frappe.render_template(
        "vciplreportsv1001/vcipl1001/report/monthwises_purchase/monthwises_purchase.html",
        {"data": data}
    )

    return {
        "labels": labels,
        "values": values,
        "html": html
    }
